plutus-ledger/src/Ledger/Index.hs

```diff
@@ -163,20 +165,22 @@ validateTransaction :: ValidationMonad m
     -> Tx
     -> m (Maybe ValidationErrorInPhase, UtxoIndex)
 validateTransaction h t = do
-    _ <- checkValidInputs (view inputsFees) t
-    (do
-        _ <- checkSlotRange h t
-        _ <- checkValuePreserved t
-        _ <- checkPositiveValues t
-        _ <- checkFeeIsAda t
+    checkSlotRange h t
+    checkValuePreserved t
+    checkPositiveValues t
+    checkFeeIsAda t
 
-        -- see note [Forging of Ada]
-        emptyUtxoSet <- reader (Map.null . getIndex)
-        unless emptyUtxoSet (checkForgingScripts t)
-        unless emptyUtxoSet (checkForgingAuthorised t)
-        unless emptyUtxoSet (checkTransactionFee t)
+    -- see note [Forging of Ada]
+    emptyUtxoSet <- reader (Map.null . getIndex)
+    unless emptyUtxoSet (checkForgingAuthorised t)
+    unless emptyUtxoSet (checkTransactionFee t)
+
+    checkValidInputs (view inputs) t
```

Comment:
I think we need to split up `checkValidInputs` into `checkValidPubKeyInputs` and `checkValidScriptInputs` - in phase 2 the script inputs should be checked